### DISCUSSION TITLE
Does pkarr can work like nomen

### DISCUSSION THREAD

**Discussion by metacatdud** [pubky/pkarr#30] [General]
I saw recently this https://nomenexplorer.com/ and thought about your project. 
Given the keep-alive feature, I think it might be used this way with no problem. 

I think is even better as you can use advertisements to change the key with another key such  as 
https://app.pkarr.org/?pk=9t8bzth7d9qng7gm3yoa165c1gge9mbe3gifaoq7bwocrojkrbsy

**mydomain** "some pub key" could rotate pretty easily to something like "some pub key 2" if we assume a client trusts "9t8bzth7d9qng7gm3yoa165c1gge9mbe3gifaoq7bwocrojkrbsy"? Right?

//EDIT add further ref: 
To give more context: https://stacker.news/items/256435

Just want to hear your thoughts. 
Cheers!

**Comment by Nuhvi** [comment]
My thoughts is that Nostr developers keep pretending that Nostr itself doesn't have a routing problem. They assume that knowing an npub is enough to find data linked to that key, ignoring that you need to also _assume_ where you should look, and that if the npub isn't published on one of the handful of popular relays, they can't be found.

So even after you spam the Bitcoin chain with your nomen transaction, resolving the actual data corresponding to that Bitcoin key is just as uncertain and relies on the good grace of the handful of popular Nostr relays.

Moreover, unique human-readable names are awful anyways and shouldn't be encouraged, because it is an artificial scarcity that has no reason to exist, instead, people should embrace the Petname system that was mentioned in the first NIP.

Pkarr is a better solution as it does not depend on any hardcoded set of relays or endpoints, and you can publish your updates from anywhere knowing that they can't be censored by any bottleneck, but unfortunately, Nostr is using secp and convincing all these client developers to adapt to using ed25519 keys as the root of identity is very hard, especially since even simple stuff like nip05 isn't supported by all clients.

So in conclusion, Pkarr is a solution to the ICANN problem, Nostr-based DNS is not, but that doesn't stop people from pretending.

That being said, if someone is willing to build Nostr client(s) that use Pkarr like you described (a better alternative to nip05) I encourage them and would support them if they needed any advice.

**Comment by metacatdud** [comment]
> My thoughts is that Nostr developers keep pretending that Nostr itself doesn't have a routing problem. They assume that knowing an npub is enough to find data linked to that key, ignoring that you need to also assume where you should look, and that if the npub isn't published on one of the handful of popular relays, they can't be found.

This is very annoying as quite often I run into inconsistencies when I see a post of a photo but no photo or missing comments etc ... With the current design I think this is very hard to solve.

> So even after you spam the Bitcoin chain with your nomen transaction, resolving the actual data corresponding to that Bitcoin key is just as uncertain and relies on the good grace of the handful of popular Nostr relays.

I thought so too. I was thinking to myself: this doesn't solve anything am I that stupid? Everyone is soo hyped.

Thanks for the reply. I think your solution super elegant. 

I am trying to build a communities (whatever communities) system. Communities make sense to me, even reddit global feed is community from a tech pov as well as nostr global feed, but I don't think (or I can't see how) you can keep those running without some serious incentives. 

I want to build a system where one can build a community this everything it needs as simple as possible and as pulggable as possible.
Create community, attach services and use it. And I think **_pkarr_** and **_ed25519_** will help me achieve the decoupling I am looking for.

I think social media/network is not going to ever pay only by itself all the classics generate the revenue from other streams and the social media is just an enabler. I think is one of the reasons nostr cannot pick up the step. Why would I run a relay? Ok, maybe if I want to host community for whatever, but even then there are far better solutions.

And last but not least, social medias have the tendency of becoming toxic. You can say one inch without being told off. As far as I am concerned  it was never my game :) I have a group of close friends, a group for family etc..

**Comment by Nuhvi** [comment]
@metacatdud I agree that social media is very hard to pull of especially that Twitter/X is already very good, but I feel that we are better of adapting Matrix protocol than reinventing it, it already has a lot of the very hard building blocks and experience building and moderating rooms.

**Reply by metacatdud** [reply]
Tried it a bit a while ago. Never quite got it 😁. It uses same principles as mastodon? As in federations. I felt is kind of mastodon only better?

**Reply by Nuhvi** [reply]
It is quite complex, but has solid sdks and many 3rd party clients. It is not the same as Mastodon, because your Home Server doesn't control what you see like Mastodon, rooms are replicated through home servers and admins control rooms.

Also, Matrix is trying to keep Ids separate from Home Servers to make migration easier, and they moved to OIDC (Oauth) for login with 3rd parties.

I don't think it will be easy to integrate Pkarr in Matrix without at least some feedback/guidance from the core devs, as I said, it is not simple.

**Comment by evd0kim** [comment]
Just leaving it here. 

> paz now supports connecting to relays that use a pkarr URL (with a z-base32 encoded public key).

https://primal.net/e/nevent1qvzqqqqqqypzqzejsjgtz7q9kd0q462nrqvpmvpmhgtgxwl5pzghh8[…]t9qqsqqdlpwnm4x7uwtcaqy7wq6k9dz7nygknk2y3kunvgx07f9ucy0cgdw935m

Sourcecode for a Nostr client based on nostralink. Uses an RDF (linked data) events store. 

https://codeberg.org/nostralink/paz

Additional info & documentation

https://pazstr.codeberg.page.

**Comment by dpc** [comment]
Nostr is is bunch of people who know very little about distributed systems and think that Bitcoin is made of pixie dust, and just borrowing tech from it magically solves problems. AFAICT, anything coming from this space technical is a waste of time.

It's somewhat useful for Bitcoiners as it's a Schelling point for a self-signed data format that is technically extremely bad, but already popular in Bitcoin-hype circles, so might as well use it. Because hype is all that maters.

>  Everyone is soo hyped.

It's all just Bitcoin-adjacent hype.

> I am trying to build a communities (whatever communities) system.

One thing that I want to point out is that decentralization is not end-user feature. Or at least it's not a feature that 99.9% of users care about.

Social networks live and die by popularity. Your tech stack and technical features don't matter. People go where other people and content is. It's virtually impossible to dethrone existing platforms with a strong network effect.

And decentralization is just making everything soo much complicated.

The phenomenon of Nostr is that is allows people who are hyped about Bitcoin LARP about using decentralized social media too, while using effectively a centralized tech pretending to do anything about decentralization. But people do care about vibes not fact, so it works. And since this group is cohesive and large enough, it's a self-sustaining group that can amuse itself with memes and zapping and marinating in own farts.

To sum up: your social network project is doomed, unless you have some non-technical reasons to exist, and even then it's probably doomed because there's already Twitter.

And don't get me started on a problem of moderation in censorship-resistant, distributed social networks...
 
> And I think pkarr and ed25519 will help me achieve the decoupling I am looking for.

Pkarr does indeed very elegantly and pragmatically solve a very important and core to building p2p systems piece. ed25519 is a solid and easy to use (but basic) cryptography.

Iroh (which builds on p2p) and Pkarr are two most enabling pieces of tech in the p2p/distributed space for me. Two pieces (p2p names, and p2p connectivity) that always seemed like insurmountable obstacles now solved by easy to use Rust libraries.

BTW. I have my own doomed to be irrelevant social platform: https://github.com/dpc/rostra using   pkarr and ed25519. It works, it's (IMO) neat, it was fun building it. But I still mostly just rant about stuff on Twitter, and Github.
